test/integration/http2_integration_test.cc

```diff
@@ -1611,6 +1611,51 @@ TEST_P(Http2FrameIntegrationTest, SetDetailsTwice) {
   EXPECT_THAT(waitForAccessLog(access_log_name_), HasSubstr("too_many_headers"));
 }
 
+TEST_P(Http2FrameIntegrationTest, OverflowingResponseCode) {
+  setDownstreamProtocol(Http::CodecClient::Type::HTTP2);
+  setUpstreamProtocol(FakeHttpConnection::Type::HTTP2);
+  // set lower outbound frame limits to make tests run faster
+  config_helper_.setOutboundFramesLimits(1000, 100);
+  initialize();
+  auto options = std::make_shared<Network::Socket::Options>();
+  tcp_client_ = makeTcpConnection(lookupPort("http"), options);
+  FakeRawConnectionPtr fake_upstream_connection;
+
+  // setup preamble and setting
+  ASSERT_TRUE(tcp_client_->write(Http2Frame::Preamble, false, false));
+  Http2Frame::SettingsFlags settings_flags = static_cast<Http2Frame::SettingsFlags>(0);
+  Http2Frame setting_frame = Http2Frame::makeEmptySettingsFrame(settings_flags);
+  ASSERT_TRUE(tcp_client_->write(std::string(setting_frame), false, false)); // empty settings
+  // Ack settings
+  settings_flags = static_cast<Http2Frame::SettingsFlags>(1); // ack
+  Http2Frame ack_frame = Http2Frame::makeEmptySettingsFrame(settings_flags);
+  ASSERT_TRUE(tcp_client_->write(std::string(ack_frame), false, false)); // ack setting
```

Comment:
First, thanks for suggesting this. This really simplified the test logic!
QQ on this: I remember you mentioning you wanted this in protocol integration, which would test all permutations. Since I now have both client and upstream logic for both h1 and h2, should I add it there?

Reply:
Btw, I tested my http2frame makeHeadersFrameWithStatus class to make sure it worked with stuff like 200 headers, it works!

Reply:
Yes, you could move it to the protocol_integration_test.cc, it could be a bit cleaner. You could use the approach I've added in PR #13635 You will need to send either H/1 response or H/2 response frame based on the upstream protocol type. Up to you if you want to do this.

Reply:
Added protocol integration.

Reply:
I also moved missing status to protocol integration :)!